### PR TITLE
[Settings] Update feature flag usage

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR moves the Settings feature flag into the correct folder and removes a check for an option value to turn the feature on. Simply updating the config will be sufficient. This is helpful so switching branches from `trunk` to any development of Settings won't require a `composer install`.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Locally, update the development config value
```diff
--- a/plugins/woocommerce/client/admin/config/development.json
+++ b/plugins/woocommerce/client/admin/config/development.json
@@ -26,7 +26,7 @@
                "printful": true,
                "remote-inbox-notifications": true,
                "remote-free-extensions": true,
-               "settings": false,
+               "settings": true,
                "shipping-label-banner": true,
                "subscriptions": true,
                "store-alerts": true,
```
2. We've renamed a file, so be sure to update composer. `cd plugins/woocommerce && composer update`
3. `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build`
4. WooCommerce > Settings
5. A placeholder for a new experience should render

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Change to underlying Settings feature usage

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/Features/Settings.php [hunk 1]
```diff
@@ -42,11 +37,6 @@ public function __construct() {
 		}
 
 		add_filter( 'woocommerce_settings_features', array( $this, 'add_feature_toggle' ) );
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Admin/Features/Settings.php]: We've removed the toggle option, should we remove this too?

**Review by ilyasfoo — COMMENTED** [review]
Testing well but minor comment

**Review by chihsuan — APPROVED** [review]
Tested well! 👍

File: plugins/woocommerce/src/Admin/Features/Settings.php (hunk 1 above)
Comment (psealock) [reply plugins/woocommerce/src/Admin/Features/Settings.php]: Good call, removed in https://github.com/woocommerce/woocommerce/pull/50977/commits/32d58de6f613defe82ee754222324410bdbe3774

**Review by ilyasfoo — APPROVED** [review]
LGTM! 🚢

**Pull request merged**